### PR TITLE
Improve documentation of tests, plans and stories

### PR DESCRIPTION
**Pull request by psss**
Document all plan attributes under a single page.
Include a test, plan and story example in the spec.
Mention core attributes in tests, plans and stories.
Improve the description of individual metadata levels.
Move the `Modules` section under the `Develop` page.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -10,27 +10,43 @@ Test Management Tool
...
+Specification
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+There are several metadata levels defined by the specification:
+
+**Core** attributes such as `summary` or `description` which are
+common across all levels are defined by the special L0 metadata.
+
+**Tests**, or L1 metadata, define attributes which are closely
+related to individual test cases such `test` script, `framework`,
```
Comment (lukaszachy) [inline README.rst:40]: How about `such -> such as`

**Review by lukaszachy — APPROVED** [review]
Looks reasonable. One nitpick, but I'm not really sure.

File: README.rst [hunk 2]
```diff
@@ -10,27 +10,43 @@ Test Management Tool
...
+directory `path` where the test should be executed, maximum test
+`duration` or packages required to run the test.
+
+**Plans**, also called L2 metadata, are used to group relevant
+tests and enable the CI. They describe how to `discover` tests for
```
Comment (thrix) [inline README.rst:45]: `enable the CI` ? maybe `enabled them in CI` ?

File: README.rst [hunk 3]
```diff
@@ -10,27 +10,43 @@ Test Management Tool
 Description
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-The ``tmt`` python module and command-line tool implement the
-Metadata Specification which allows storing all needed test
-execution data directly within a git repository. In this way, it
-makes testing independent on any external test management system.
+The ``tmt`` tool provides a user-friendly way to work with tests.
+It allows to comfortably create new tests, safely and easily run
```
Comment (hegerj) [inline README.rst:14]: I don't believe `allow to verb` is grammatically correct, it should be something like
`allows comfortably creating`
Source: https://english.stackexchange.com/questions/60271/grammatical-complements-for-allow/60285#60285

File: README.rst [hunk 4]
```diff
@@ -10,27 +10,43 @@ Test Management Tool
...
+tests across different environments, review test results, debug
+test code and enable tests in the CI using a consistent and
+concise config.
+
+The python module and command-line tool implement the Metadata
+Specification which allows to store all needed test execution data
```
Comment (hegerj) [inline README.rst:20]: Same as above,
`aloows storing`

File: docs/spec.rst [hunk 5]
```diff
@@ -30,16 +30,17 @@ Level 1: Tests
...
 
 Level 2: Plans
-    This level represents :ref:`/spec/plans` made up of individual
-    :ref:`/spec/steps` describing how to
-    :ref:`/spec/steps/provision` the environment for testing and
-    how to :ref:`/spec/steps/prepare` it or which frameworks
-    should be used to :ref:`/spec/steps/execute` tests relevant
-    for given :ref:`/spec/plans/context`.
+    :ref:`/spec/plans` are used to group relevant tests and enable
+    the CI. They describe how to :ref:`/spec/plans/discover` tests
```
Comment (thrix) [inline docs/spec.rst:34]: same ask to rephrase

**Review by thrix — APPROVED** [review]
LGTM, just one small thing found

File: spec/plans/main.fmf [hunk 6]
```diff
@@ -1,33 +1,60 @@
...
-    * CI system is configured by ``default`` to use compose ``C1`` installed on machines with ``M1`` GB of memory.
-    * When inspecting an artifact CI can ``detect`` that for this particular build target, compose ``C2`` is a much better choice, and memory is fine as it is.
-    * User can still explicitly ``define`` in the configuration that at least ``M2`` GB of memory is needed for successful test execution which overrides both default and detected values.
+    **Plans**, also called L2 metadata, are used to group relevant
+    tests and enable the CI. They describe how to `discover` tests
+    for execution, how to `provision` the environment, how to
+    `prepare` it for testing, how to `execute` tests, `report`
+    results and finally how to `finish` the test job.
+
+    Each of the six steps mentioned above supports multiple
```
Comment (hegerj) [inline spec/plans/main.fmf:12]: supports -> support

File: spec/plans/main.fmf [hunk 7]
```diff
@@ -1,33 +1,60 @@
...
+    implementations.  The default methods are listed below.
+
+     * :ref:`/spec/plans/discover`: shell
+     * :ref:`/spec/plans/provision`: virtual
+     * :ref:`/spec/plans/prepare`: shell
+     * :ref:`/spec/plans/execute`: tmt
+     * :ref:`/spec/plans/report`: display
+     * :ref:`/spec/plans/finish`: shell
+
+    Thanks to clearly separating test steps it is possible to run
```
Comment (hegerj) [inline spec/plans/main.fmf:22]: separating -> separated?

**Review by hegerj — CHANGES_REQUESTED** [review]

File: spec/plans/main.fmf (hunk 6 above)
Comment (hegerj) [reply spec/plans/main.fmf:12]: I take it back, `supports` is correct :)

**Comment by psss** [comment]
Thanks for the review, all comments should be addressed.

**Review by hegerj — APPROVED** [review]

**Pull request merged**